docs/notebooks/active_learning.pct.py

```diff
@@ -0,0 +1,177 @@
+# %% [markdown]
+# # Active Learning
+
+# %% [markdown]
+# Sometimes rather than optimizing a black-box function, we want to learn the function better by reducing our model uncertainty. The one way to do this is by doing sampling in unknown inputs regions (active learning). This notebook demonstrates to perform Bayesian active learning using `trieste`.
+
+# %%
+# %matplotlib inline
+import numpy as np
+import tensorflow as tf
+import pandas as pd
+
+np.random.seed(1793)
+tf.random.set_seed(1793)
+
+# %% [markdown]
+# ## Describe the problem
+#
+# In this example, we will consider doing active learning for the log branin function.
+
+
+# %%
+from trieste.utils.objectives import branin
+from util.plotting_plotly import plot_function_plotly
+from trieste.space import Box
+
+
+def log_branin(x):
+    return tf.math.log(branin(x))
+
+
+search_space = Box([0, 0], [1, 1])
+
+fig = plot_function_plotly(log_branin, search_space.lower, search_space.upper, grid_density=20)
+fig.update_layout(height=400, width=400)
+fig.show()
+
+# %% [markdown]
+# We begin our Bayesian active learning steps after evaluating two initial points sampled by Halton sequence.
+
+# %%
+import trieste
+
+observer = trieste.utils.objectives.mk_observer(log_branin)
+
+num_initial_points = 2
+initial_query_points = search_space.sample_halton(num_initial_points)
+initial_data = observer(initial_query_points)
+
+
+# %% [markdown]
+# ## Surrogate model
+#
+# Just like in sequential optimization, we fit a surrogate Gaussian process model to the initial data.
+
+# %%
+import gpflow
+
+
+def build_model(data):
+    variance = tf.math.reduce_variance(data.observations)
+    kernel = gpflow.kernels.RBF(variance=variance, lengthscales=[0.2, 0.2])
+    gpr = gpflow.models.GPR(data.astuple(), kernel, noise_variance=1e-5)
+    gpflow.set_trainable(gpr.likelihood, False)
+
+    return {
+        "model": gpr,
+        "optimizer": gpflow.optimizers.Scipy(),
+        "optimizer_args": {
+            "minimize_args": {"options": dict(maxiter=100)},
+        },
+    }
+
+
+model = build_model(initial_data)
+
+# %% [markdown]
+# ## Active learning using predictive variance
```

Comment:
Can you add a ref here?

Reply:
Couldnt find any paper that focuses on predictive variance
one candidate is this paper https://arxiv.org/pdf/0805.4359.pdf on chapter 2.2.1, they mention this a litle bit.
WDYT?

Reply:
@henrymoss  any comment on this?

Reply:
Possibly it's simple enough that we don't need a ref. In the case of batch > 1, I don't think that using the determinant of the covmatrix is standard and we could use a reference.

Reply:
maybe use 
MacKay, D. J. (1992). Information-based objective functions for active data selection. Neural computation, 4(4), 590-604.
instead?

Reply:
I don't see the MacKay reference here in the text?